src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/OptionsTests.cs

```diff
@@ -475,11 +475,18 @@ public static void Options_JsonSerializerContext_DoesNotFallbackToReflection()
         }
 
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
-        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        public static void Options_JsonSerializerContext_GetConverter_DoesNotFallBackToReflectionConverter()
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [InlineData(false)]
+        [InlineData(true)]
+        public static void Options_JsonSerializerContext_GetConverter_DoesNotFallBackToReflectionConverter(bool isCompatibilitySwitchExplicitlyDisabled)
         {
-            RemoteExecutor.Invoke(static () =>
+            RemoteExecutor.Invoke(static (string isCompatibilitySwitchExplicitlyDisabled) =>
             {
+                if (bool.Parse(isCompatibilitySwitchExplicitlyDisabled))
```

Comment:
Nit: this can be done through the options property like this:

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Runtime/tests/System/UseResourceKeysTest.cs#L14-L15